packages/components/src/internal/components/editable/EditableGrid.tsx

```diff
@@ -1006,13 +1068,21 @@ export class EditableGrid extends PureComponent<EditableGridProps, EditableGridS
         return Promise.resolve();
     };
 
-    bulkUpdate = async (updatedData: OrderedMap<string, any>): Promise<void> => {
-        const { editorModel, queryInfo, onChange } = this.props;
+    bulkUpdate = async (updatedData: OrderedMap<string, any>): Promise<Partial<EditorModelProps>> => {
+        const { editorModel, queryInfo, onChange, bulkUpdateProps } = this.props;
+        if (!updatedData) return Promise.resolve(undefined);
+
         const selectedIndices = this.getSelectedRowIndices();
-        const editorModelChanges = await updateGridFromBulkForm(editorModel, queryInfo, updatedData, selectedIndices);
+        const editorModelChanges = await updateGridFromBulkForm(
+            editorModel,
+            queryInfo,
+            updatedData,
+            selectedIndices,
+            bulkUpdateProps?.excludeRowIdx
+        );
         onChange(editorModelChanges);
         // The result of this promise is used by toggleBulkUpdate, which doesn't expect anything to be passed
-        return Promise.resolve();
+        return Promise.resolve(editorModelChanges);
```

Comment:
I see that returning `Promise.resolve()` was already here but this just seems really weird. The comment does not seem to be true anymore, at least with regards to `toggleBulkUpdate`, so I'm not sure we need all this returning of a resolved promise. We can just return `editorModelChanges` and it will be promised.

Reply:
Because this.bulkUpdate is still used for BulkAddUpdateForm.onSubmitForEdit, it needs to remain a promise.